litellm/router_utils/client_initalization_utils.py

```diff
@@ -303,6 +315,7 @@ def set_client(litellm_router_instance: LitellmRouter, model: dict):
                     "azure_endpoint": api_base,
                     "api_version": api_version,
                     "azure_ad_token": azure_ad_token,
+                    "azure_ad_token_provider": azure_ad_token_provider,
```

Comment:
can we add a basic unit test here - https://github.com/BerriAI/litellm/blob/04d69464e2c98f62569ef3a108bbc7aecc4ac8a1/litellm/tests/test_router_client_init.py#L17

with a mock test to make sure there's no future regressions

Reply:
1. OK, should I make another test-function next to this one with mocking environment variables (`AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET` and `AZURE_TENANT_ID`)?
2. The test you mentioned is skipped on my machine (problably due to the missing `pytest-asyncio` plugin and `pytest.mark.asyncio` decorator). Is this expected behaviour and this would be executed on CI/CD?

Reply:
Prepared the commit [85a8b9e](https://github.com/BerriAI/litellm/pull/5131/commits/85a8b9ee3db14ed7537d7ff75807867e151f01dc) with environment variables and other prerequisites mocked.